ste/jobStatusManager.go

```diff
@@ -67,18 +97,32 @@ func (jm *jobMgr) handleStatusUpdateMessage() {
 	js.JobID = jm.jobID
 	js.CompleteJobOrdered = false
 	js.ErrorMsg = ""
+	allXferDoneHandled := false
 
 	for {
 		select {
-		case msg := <-jstm.partCreated:
+		case msg, ok := <-jstm.partCreated:
```

Comment:
Once channel closed, this for loop keep running as <-jstm.partCreated always true. That's unnecessary cpu cycles. wasted.  It will delay the processing of other messages.